core/src/main/java/org/elasticsearch/index/query/QueryBuilder.java

```diff
@@ -40,6 +43,55 @@
  */
 public abstract class QueryBuilder<QB extends QueryBuilder> extends ToXContentToBytes implements NamedWriteable<QB> {
 
+    public static enum Operator {
```

Comment:
maybe we could make the enum implement Writeable<Operator> , remove static from readFrom, and add a PROTOTYPE constant to use for deserialization as we do with queries?